driver/linux/sgx_main.c

```diff
@@ -309,14 +248,16 @@ static int sgx_pm_suspend(struct device *dev)
 static SIMPLE_DEV_PM_OPS(sgx_drv_pm, sgx_pm_suspend, NULL);
 
 
-#ifdef CDEV_BUILD
 static struct bus_type sgx_bus_type = {
 	.name	= "sgx",
 };
 
 struct sgx_context {
 	struct device dev;
 	struct cdev cdev;
+	struct device provision_dev;
+	struct cdev provision_cdev;
+	struct kobject *kobj_dir;
```

Comment:
Can we just attach the driver info to the driver/device node that's already in sysfs, e.g /bus/sgx or /module/intel_sgx?
The top dir of sys seems only contain top categories of things like bus, block, class, etc.

Reply:
The proposed path is /sys/kernel/sgx/, I think /module will be different between the OOT and DCAP driver.